File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& v) {
        
        if(v.size() == 1)
        {
            v[0] = -1;
            return v;
        }else if(v.size() == 2)
        {
            v[0] = v[1];
            v[1] = -1;
            return v;
        }
            
        for(int i = 0 ; i < v.size() ; i++)
        {
            v[i] = *max_element(v.begin() + (i+1) , v.end());
        }
        v[v.size()-1] = -1;
        
        return v;
        
    }
};

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    let max = -1;
    for(let i = arr.length - 1; i >= 0 ; i--){
        const current = arr[i];
        arr[i] = max;
        max = Math.max(max,current);
    }
    return arr
};